tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/stomp/StompTest.java

```diff
@@ -1501,6 +1501,26 @@ public void testSubscribeToTopicWithNoLocal() throws Exception {
       conn.disconnect();
    }
 
+   @Test
+   public void testSubscribeToTopicWithNoLocalAndNormal() throws Exception {
+      conn.connect(defUser, defPass);
+      subscribeTopic(conn, RandomUtil.randomString(), null, null, true, true);
+      subscribeTopic(conn, RandomUtil.randomString(), null, null, true, false);
+      int normalCount = 0;
+      int noLocalCount = 0;
+      for (Binding binding : server.getPostOffice().getBindingsForAddress(SimpleString.of(getTopicPrefix() + getTopicName())).getBindings()) {
+         if (binding.getFilter() != null && binding.getFilter().getFilterString().toString().contains("__AMQ_CID")) {
+            noLocalCount++;
+         } else {
+            normalCount++;
+         }
+      }
+      assertEquals(1, noLocalCount);
+      assertEquals(1, normalCount);
+
+      conn.disconnect();
```

Comment:
Probably worth verifying removals behave are as expected also, and perhaps sending a message after and checking the expected behaviours, e.g around the header.

Reply:
I think I got all the relevant cases.